modern/src/common/localization.js

```diff
@@ -128,8 +131,40 @@ for (let i = 0; i < languages.length; i++) {
   }
 }
 
-const selectedLanguage = supportedLanguages[language];
+let selectedLanguage = supportedLanguages[language];
+
+export const findStringKeys = (predicate) => {
+  return Object.keys(selectedLanguage.data).filter(predicate);
+}
+
+export default key => {
+  return selectedLanguage.data[key];
+};
+
+const setSelectedLanguage = (language) => {
+  selectedLanguage = supportedLanguages[language];
+  localStorage.setItem('language', language);
+}
+
+const defaultLanguage = language;
+
+const LocalizationContext = createContext({
+  language
+});
 
-export const findStringKeys = (predicate) => Object.keys(selectedLanguage.data).filter(predicate);
+export function LocalizationProvider(props) {
+  const [language, setLanguage] = useState(defaultLanguage);
+
+  const handleLanguageChange = (nextLanguage) => {
+    setSelectedLanguage(nextLanguage);
+    setLanguage(nextLanguage);
+  };
+
+  return <LocalizationContext.Provider value={{ language, setLanguage: handleLanguageChange, languageList }}>
+    {props.children}
+  </LocalizationContext.Provider>
+}
 
-export default (key) => selectedLanguage.data[key];
+export const useLocalization = () => {
+  return useContext(LocalizationContext);
+}
```

Comment:
I have looked through the code here and it seems like it's a mix of global state and react. I think it should be simplified and streamlined to be a proper react context.

Reply:
Can you give me an example some where in the project with the pattern you mean or explain more please?

Reply:
We don't use it in the project currently, but you should be able to find some examples on the official React website.